src/Entities/Player.cpp

```diff
@@ -1861,6 +1863,17 @@ void cPlayer::SetVisible(bool a_bVisible)
 
 
 
+void cPlayer::PermuteEnchantmentSeed()
+{
+	// Get the next rand uint and use that as the next seed
+	MTRand Random(m_EnchantmentSeed);
+	m_EnchantmentSeed = Random.RandInt<unsigned int>();
```

Comment:
Why do we need to make a new MTRand seeded with m_EnchantmentSeed, couldn't we just make a new random seed from GetRandomProvider()?

Reply:
We can and probably should used GetRandomProvider. I had misunderstood the enchanting mechanics when I wrote that.